File: SmartWarehouse_Project/Entities/Company.cs
namespace SmartWarehouse.Entities;

public class Company
{
    public int Id { get; set; }
    public Guid CompanyId { get; set; }

    public string Name { get; set; } = string.Empty;
    public string TaxNumber { get; set; } = string.Empty;
    public string Email { get; set; } = string.Empty;
    public string Phone { get; set; } = string.Empty;
    public string Address { get; set; } = string.Empty;

    public bool IsActive { get; set; }
    public bool IsDeleted { get; set; }

    public DateTime CreatedAt { get; set; }
    public DateTime? UpdatedAt { get; set; }
}

File: smart-warehouse-ui/src/types/company.ts
export interface Company {
  id: number;
  companyId: string;
  name: string;
  taxNumber: string;
  email: string;
  phone: string;
  address: string;
  isActive: boolean;
}

export interface CreateCompanyDto {
  companyId: string;
  name: string;
  taxNumber: string;
  email: string;
  phone: string;
  address: string;
  isActive: boolean;
}

export interface UpdateCompanyDto {
  companyId: string;
  name: string;
  taxNumber: string;
  email: string;
  phone: string;
  address: string;
  isActive: boolean;
}

export interface PagedResult<T> {
  success: boolean;
  data: T[];
  totalCount: number;
  page: number;
  pageSize: number;
  totalPages: number;
}
